pkg/content/content.go

```diff
@@ -0,0 +1,53 @@
+package content
+
+import (
+	"strings"
+
+	"github.com/readium/go-toolkit/pkg/content/element"
+	"github.com/readium/go-toolkit/pkg/content/iterator"
+)
+
+type Content interface {
+	Text(separator *string) (string, error) // Extracts the full raw text, or returns null if no text content can be found.
+	Iterator() iterator.Iterator            // Creates a new iterator for this content.
+	Elements() ([]element.Element, error)   // Returns all the elements as a list.
```

Comment:
In the Kotlin toolkit, `text()` and `elements()` are in the interface with a default implementation, but they should probably be extensions instead. The only thing that should be required to implement is `Iterator`.

Maybe we can remove them from the interface and keep only `ContentText()` and `ContentElements()` on the side?

Alternatively if we want to keep the API easy to use, `Content` could be a struct and have a `IteratorFactory` interface instead.